### PR TITLE
Move theme dataset attributes to the body element and namespace them

### LINKED ISSUE #6554: Move to `data-theme-light` to the document body
Right now we have a `data-theme-light` HTML attribute on the `div.jp-LabShell` that is a child of the document body. Because we attach dialogs to the document body, this theming hint isn't available to style the dialog. The proposed solution is to move this attribute to the document body. There are also other UI elements (completer) that we attach to the body that might need access to this attribute.

### PR DESCRIPTION
**Pull request by jasongrout**
## References

Fixes #6554 

## Code changes

This moves the theme variables to the document.body element from the app.shell element.

I also audited where we are doing things with document.body, and as a result this also moves several other attributes from document.body to the app.shell element.

## Backwards-incompatible changes

The theme dataset variables change names.

### CODE REVIEW

**Comment by jasongrout** [comment]
@ian-r-rose - the code to move widgets between sidebars seems to use a body attribute solely as a way of having global state. Can we change the code to not use the body element as a place to hold global state? Right now this PR is a little broken, in that I just commented out relevant code there.

**Comment by ian-r-rose** [comment]
The sidebar dataset was mostly not used (and has maybe been in lab for a long time) -- I think that the usage you pointed to was just a fallback in case the context-menu-widget-finding failed.

File: packages/application-extension/src/index.tsx [hunk 1]
```diff
@@ -415,11 +415,11 @@ const sidebar: JupyterFrontEndPlugin<void> = {
           } else {
             side = 'left';
           }
-        } else if (document.body.dataset.leftSidebarWidget) {
-          id = document.body.dataset.leftSidebarWidget;
+        } else if (app.shell.dataset.leftSidebarWidget) {
```
Comment (ian-r-rose) [inline packages/application-extension/src/index.tsx]: I have no objection to removing `left-sidebar-widget` and `right-sidebar-widget` from the document body, but then we should remove these checks and exit this command early if no sidebar widget is found.

**Comment by jasongrout** [comment]
Thanks! This looks good to me too.

File: packages/application-extension/src/index.tsx (hunk 1 above)
Comment (jasongrout) [reply packages/application-extension/src/index.tsx]: Looks good.

**Pull request merged**